external-resolver/src/main/scala/org/genivi/sota/resolver/DeviceDataMigrator.scala

```diff
@@ -36,5 +36,11 @@ object DeviceDataMigrator {
       now               = Instant.now()
       nonForeignPkgs    = nonForeignPkgRows.map{ r => InstalledForeignPackage(r._2, r._3, now) }
       foreignPkgs       <- ForeignPackages.listPackages
-    } yield foreignPkgs ++ nonForeignPkgs
+    } yield {
+      //account for the fact that some package names are capitalised and some aren't
+      val foreignPkgIds = foreignPkgs.map(r => r.packageId.mkString)
+      foreignPkgs ++ nonForeignPkgs.filter { p =>
+        foreignPkgIds.count(pkg => p.packageId.mkString.equalsIgnoreCase(pkg)) == 0
```

Comment:
Don't forget that an `InstalledForeignPackage` also have a `device`-uuid that you much check for.